### ISSUE TITLE
WAU tries to update uninstalled apps

### ISSUE THREAD

**Issue by ppvnf** [Romanitho/Winget-AutoUpdate#236]
This problem first happened with Whatsapp, so I've blacklisted it. Today WAU tried to update Playnite. Neither of them are actually installed (although I had installed them in the past).
[updates.log](https://github.com/Romanitho/Winget-AutoUpdate/files/10021601/updates.log)

**Comment by KnifMelti** [comment]
https://github.com/microsoft/winget-pkgs/blob/master/manifests/p/Playnite/Playnite/10.7/Playnite.Playnite.installer.yaml
Scope: user

This is where winget checks if it's installed or not:
HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\Playnite_is1

You could check if that reg exists and if it does delete it.

**Comment by ppvnf** [comment]
> You could check if that reg exists (then it's installed in user context) and if it does delete it (or uninstall from Control Panel).

"HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\Playnite_is1" does not exist nor Playnite is in the Control Panel uninstall list

**Comment by KnifMelti** [comment]
My best shot...

**Comment by ppvnf** [comment]
> My best shot...

Would it be helpful or a complete waste of time if I installed an older version of Playnite on a VM with WAU and then uninstalled it before running WAU?

**Comment by KnifMelti** [comment]
I installed Playnite older vesion, with **WingetUI** - it installed in **user** context - ran a WAU check and it upgraded in user context.
Uninstalled via [winget-installed](https://github.com/Romanitho/Winget-Install) and it's gone,
WAU doesn't detect it.
In your logs WAU detects it installed in **System** context...

**Comment by KnifMelti** [comment]
Have you checked under **HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall** if **Playnite** exists there?

**Comment by ppvnf** [comment]
> Have you checked under **HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall** if **Playnite** exists there?

nope

**Comment by KnifMelti** [comment]
If you start a cmd and runs `winget upgrade --source winget`, is Playnite in the list?
This is the command **WAU** uses to find upgrades needed.

**Comment by ppvnf** [comment]
> If you start a cmd and run `winget upgrade --source winget`, is Playnite in the list? This is the command **WAU** uses to find upgrades needed.

nope, I also tried `winget list --source winget` with no success

**Comment by KnifMelti** [comment]
Then there's a Ghost in your Machine...

**Comment by ppvnf** [comment]
> ...or Romanitho can find out (maybe in the comparison between System/User installs)?

Tbh I don't actually know what is System/User install, I just click next when installing manually or use default option with Winget/WingetUI... Thanks for trying to figure it out tho

**Comment by KnifMelti** [comment]
An user install is  when you as a user can install it in your own context (without being admin), in winget .yaml.terms: **Scope: user**:
https://github.com/microsoft/winget-pkgs/blob/master/manifests/p/Playnite/Playnite/10.7/Playnite.Playnite.installer.yaml
A system install is when it installs to every user on the machine (requires admin rights).

**Comment by ppvnf** [comment]
I might have accidentally installed or uninstalled Playnite as admin

**Comment by KnifMelti** [comment]
Can you in WingetUI find it under:
![image](https://user-images.githubusercontent.com/102996177/202776223-567b7814-e768-4693-a957-c496fd1822ab.png)

**Comment by ppvnf** [comment]
I've it "installed" with Scoop (although it's a portable version and doesn't show in ctrl panel), but the Winget package is also no longer shown there

**Comment by KnifMelti** [comment]
Hmm, **Scoop** Portable in the play then.. ..no more idea from me 🤯

**Comment by KnifMelti** [comment]
Last try:
C:\Users\%USERNAME%\scoop\apps
Is **Playnite** there?

**Comment by ppvnf** [comment]
> Last try: `C:\Users\%USERNAME%\scoop\apps` Is **Playnite** there?

Yes, but it is the portable version

**Comment by KnifMelti** [comment]
I've tried everything from installing different **Playnite** versions via Winget/Scoop/Web (as admin and std. user), but I can't replicate the problem with **WAU** wanting to upgrade it when it's not installed.
Ghost in the Machine...
...but, you're satisfied with **WAU** otherwise, thanks to @Romanitho?

**Comment by ppvnf** [comment]
> I've tried everything from installing different **Playnite** versions via Winget/Scoop/Web (as admin and std. user), but I can't replicate the problem with **WAU** wanting to upgrade it when it's not installed. Ghost in the Machine... ...but, you're satisfied with **WAU** otherwise, thanks to @Romanitho?

Yes, I am pretty satisfied with WAU otherwise. I've had this problem only 2 times (with Playnite and Whatsapp) and it's pretty easy to workaround it by blacklisting the affected packages

**Comment by ppvnf** [comment]
> Hi :) Is scoop able to uninstall apps ? This only way to uninstall properly I guess, is to uninstall it in system context.

Yes, Scoop version is just a portable one (it's not intalled in the context menu). You can only uninstall it by deleting manually the "install" folder or using `scoop uninstall`. Maybe there are some leftover registry entries...  Anyway, I've decided to reinstall Windows since my install was kinda old (3 years) and was getting a little bit slow. Hopefully this problem will also be fixed

**Comment by Romanitho** [comment]
Ok, good idea ;)
